### PR TITLE
Update sarama and goavro dependencies

### PR DESCRIPTION
**Pull request by shermanCRL**
Just a deps update, not driven by a particular issue, but to not be a year behind in the next release.

Brings along updates to x/net, x/sync, xdg-go/scram and other indirect dependencies.

Addresses https://github.com/cockroachdb/cockroach/issues/81907

Release note: Avro decimal encoding is now higher precision, which we believe to be an improvement/bug fix.

Jira: none
Epic: none

### CODE REVIEW

File: DEPS.bzl [hunk 1]
```diff
@@ -2365,11 +2365,9 @@ def go_deps():
         name = "com_github_eapache_go_resiliency",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/eapache/go-resiliency",
-        sha256 = "39333303f947a85e0c35e9969d56e05776034b1ae91e75cbf9211ead5870d982",
-        strip_prefix = "github.com/eapache/go-resiliency@v1.2.0",
-        urls = [
-            "https://storage.googleapis.com/cockroach-godeps/gomod/github.com/eapache/go-resiliency/com_github_eapache_go_resiliency-v1.2.0.zip",
-        ],
+        # TODO: mirror this repo (to fix, run `./dev generate bazel --mirror`)
```
Comment (RaduBerinde) [inline DEPS.bzl:2368]: Looks like you need to run this

File: DEPS.bzl [hunk 2]
```diff
@@ -6883,6 +6875,14 @@ def go_deps():
             "https://storage.googleapis.com/cockroach-godeps/gomod/github.com/pierrec/lz4/com_github_pierrec_lz4-v2.6.0+incompatible.zip",
         ],
     )
+    go_repository(
```
Comment (RaduBerinde) [inline DEPS.bzl:6878]: I wonder if we really need both of this (maybe we can update alll code to use v4)?

Comment (shermanCRL) [reply DEPS.bzl:6886]: Looks like only one place for the old package, and recent: https://github.com/shermanCRL/cockroach/blob/master/pkg/sql/sem/builtins/compression.go cc @miretskiy

Comment (shermanCRL) [reply DEPS.bzl:6886]: I’ve removed the older one.

**Comment by shermanCRL** [comment]
Here’s are some diff:
- https://github.com/jcmturner/gokrb5/compare/v8.4.2...v8.4.3
- https://github.com/golang/net/compare/a158d28d115b...f486391704dc
- https://github.com/golang/sync/compare/886fb9371eb4...7f9b1623fab7

**Comment by shermanCRL** [comment]
Avro encoding q for @HonoreDB.

After upgrading the Avro library, `TestAvroLedger` fails due to decimal encoding.

Before, it was `"bytes.decimal":"44061/500"` and after is `"bytes.decimal":"88122259/1000000"`. Nearly the same number, the latter is more precise.

This seems strictly better, as it implies that the status quo is truncating the precision.  Do you see this as a breaking change, or any other downsides?

I assume that any downstream Kafka (or any Avro consumer) will succeed. I’ll probably call it out as a breaking change for documentation purposes, just in case.

**Comment by knz** [comment]
Thanks I have looked at the first 3:

- https://github.com/jcmturner/gokrb5/compare/v8.4.2...v8.4.3

  1) there are two release note-worthy bug fixes in there. 
       - ["if we don't know the SPN's realm, ask the client realm's KDC"](https://github.com/jcmturner/gokrb5/commit/1150460329ccca6829d243c06413a8182c40e29e)
       - case-insensitive principal name matching https://github.com/jcmturner/gokrb5/commit/ae7841b69ddeb2280adbc16624abfca094a9b351
  2) there is a suspicious change which we'll want @catj-cockroach 's eyes on: https://github.com/jcmturner/gokrb5/commit/a037a7c28a3b0a9a09a4c9216d9fd18b8f2417b6

- https://github.com/golang/net/compare/a158d28d115b...f486391704dc

  The HTTP2 stream windowing behavior is being updated. Since this code is underpinning all our RPC infrastructure, we will need to be extra mindful of perf/stability regressions and look at this again if there's concerns. But that doesn't block this PR.

- https://github.com/golang/sync/compare/886fb9371eb4...7f9b1623fab7

  Nothing to see there


The sarama and avro dep updates are owned by the CDC team. I don't have anything to say about those. Looks like you've already identified the main change we'd need to announce.

**Comment by HonoreDB** [comment]
>Do you see this as a breaking change, or any other downsides?
This looks like a bug fix to me! It is a potential breaking change because most avro changes are breaking changes, even purely additive ones, but with docs it seems fine.

**Comment by shermanCRL** [comment]
On Kerberos, specifically our update of https://github.com/jcmturner/gokrb5: I do not find any direct use in our codebase, only that it’s a dependency of Sarama.

@HonoreDB do you know anything about Kerberos & Sarama & Kafka? Is it a facility available to our users, or somehow part of Kafka’s protocols?

**Comment by HonoreDB** [comment]
> On Kerberos, specifically our update of https://github.com/jcmturner/gokrb5: I do not find any direct use in our codebase, only that it’s a dependency of Sarama.
> 
> @HonoreDB do you know anything about Kerberos & Sarama & Kafka? Is it a facility available to our users, or somehow part of Kafka’s protocols?
> 
> I assume this is the place where it would appear if we were offering it? https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/changefeedccl/sink_kafka.go#L796-L944
> 
> Would you also please double-check my assertion that there is no direct use elsewhere?

We don't currently have a way (that I know of) to pipe through kerberos authentication config to Sarama through a create changefeed command, even though Sarama supports it, so no impact on changefeeds. I do think we use this package in our CLI tool though to support kerberos authentication through https://github.com/otan/gopgkrb5 .

**Comment by shermanCRL** [comment]
@HonoreDB nice spot. I added @otan as a reviewer. Oliver, perhaps you may wish to update the dependency of https://github.com/otan/gopgkrb5 on https://github.com/jcmturner/gokrb5/compare/v8.4.2...v8.4.3 while we are doing this?

**Comment by otan** [comment]
> perhaps you may wish to update the dependency

done.

**Comment by shermanCRL** [comment]
Hi @catj-cockroach, would you offer your review of the Kerberos changes that knz [mentions above](https://github.com/cockroachdb/cockroach/pull/95544#issuecomment-1398326903)? I don’t know enough about Kerberos to say if those changes are interesting or risky:

> - https://github.com/jcmturner/gokrb5/compare/v8.4.2...v8.4.3
> 
>   1) there are two release note-worthy bug fixes in there. 
>        - ["if we don't know the SPN's realm, ask the client realm's KDC"](https://github.com/jcmturner/gokrb5/commit/1150460329ccca6829d243c06413a8182c40e29e)
>        - case-insensitive principal name matching https://github.com/jcmturner/gokrb5/commit/ae7841b69ddeb2280adbc16624abfca094a9b351
>   2) there is a suspicious change which we'll want @catj-cockroach 's eyes on: https://github.com/jcmturner/gokrb5/commit/a037a7c28a3b0a9a09a4c9216d9fd18b8f2417b6

As far as I know, [this](https://www.cockroachlabs.com/docs/stable/gssapi_authentication.html) is the primary place the Kerberos is exposed to end users. Thanks.

**Comment by shermanCRL** [comment]
For the sake of expediency, I’ve taken Kerberos out of scope here by updating to an earlier version of Sarama. Looks like we are therefore in good shape with the above diff reviews. @HonoreDB @knz if you agree, would you approve? Thanks.

**Review by knz — APPROVED** [review]

**Pull request merged**